### ISSUE TITLE
format_test.format_double fails

### ISSUE THREAD

**Issue by UnitedMarsupials** [fmtlib/fmt#3760]
Compiling on FreeBSD-14/amd64, using the base compiler (clang-16.0.6). The build succeeds, but one of the tests fails:
```
[ RUN      ] format_test.format_double
.../fmtlib/work/fmt-10.1.1/test/format-test.cc:1354: Failure
Expected equality of these values:
  fmt::format("{:.9a}", xd)
    Which is: "0x2.000000000p+2"
  buffer
    Which is: "0x1.000000000p+3"
[  FAILED  ] format_test.format_double (3 ms)
```
Please, advise. Thank you!

**Comment by phprus** [comment]
The error occurs due to different rounding in different libc (macOS, glibc, FreeBSD 13,14):

| format      | fmt                   | glibc                 |
|-------------|-----------------------|-----------------------|
| ``{:.10a}`` | ``0x1.ffffffffffp+2`` | ``0x1.ffffffffffp+2`` |
| ``{:.9a}``  | ``0x2.000000000p+2``  | ``0x2.000000000p+2``  |

Previous issues: #3512, #3433

I will make a fix for these tests.

**Comment by phprus** [comment]
Fix for test: #3762
